cosi/ChartGenerator/components/ChartGenerator.vue

```diff
@@ -0,0 +1,447 @@
+<script>
+/* eslint-disable vue/no-unused-components */
+import Vue from "vue";
+import Tool from "../../../../src/modules/tools/Tool.vue";
+import {mapGetters, mapMutations} from "vuex";
+import getters from "../store/gettersChartGenerator";
+import mutations from "../store/mutationsChartGenerator";
+import {scaleLinear} from "d3-scale";
+import {color} from "d3-color";
+import LineChart from "./charts/LineChart.vue";
+import BarChart from "./charts/BarChart.vue";
+
+export default {
+    name: "ChartGenerator",
+    components: {
+        Tool,
+        LineChart,
+        BarChart
+    },
+    data () {
+        return {
+            test: 0,
+            dataSets: [],
+            activeGraph: 0,
+            newType: "BarChart",
+            forceGraphUpdate: 1
+        };
+    },
+    computed: {
+        ...mapGetters("Tools/ChartGenerator", Object.keys(getters))
+    },
+    watch: {
+        newDataSet (dataSet) {
+            dataSet.cgid = dataSet.id + "-" + dataSet.name;
+
+            const checkDouble = this.dataSets.find(x => x.cgid === dataSet.cgid);
+
+            if (checkDouble) {
+                const index = this.dataSets.indexOf(checkDouble);
+
+                this.dataSets.splice(index, 1);
+                this.dataSets.push(dataSet);
+            }
+            else {
+                this.dataSets.push(dataSet);
+            }
+            if (dataSet.target === "" || dataSet.target === undefined || dataSet.target === null) {
+                this.$store.commit("Tools/ChartGenerator/setActive", true);
```

Comment:
über ...mapMutations einbinden

Reply:
Ach is ja schon, s. Zeile 70/77. also kannstes mit "this.setActive(true)" aufrufen